packages/camera/camera/lib/camera.dart

```diff
@@ -560,11 +419,8 @@ class CameraController extends ValueNotifier<CameraValue> {
       );
     }
     try {
+      await CameraPlatform.instance.stopVideoRecording(_cameraId);
       value = value.copyWith(isRecordingVideo: false);
-      await _channel.invokeMethod<void>(
-        'stopVideoRecording',
-        <String, dynamic>{'textureId': _textureId},
-      );
     } on PlatformException catch (e) {
       throw CameraException(e.code, e.message);
     }
```

Comment:
Make sure the `isRecordingVideo` is updated to `false` when the exception occurs:

```suggestion
    } on PlatformException catch (e) {
      value = value.copyWith(isRecordingVideo: false);
      throw CameraException(e.code, e.message);
    }
```